Framework/PythonInterface/mantid/utils/dgs/_geometry.py

```diff
@@ -0,0 +1,268 @@
+    3. If goniometer_constraints is False, goniometer_range is ignored
+    """
+    # check input parameters
+    try:
+        len_hkl = len(hkl)
+        if len(hkl[0]) != 3:
+            raise ValueError()
+    except (TypeError, ValueError):
+        raise ValueError('hkl is not an array of triplets')
+
+    try:
+        # check if float
+        Ei = float(Ei)
+        Ei = np.full(len_hkl, Ei)
+    except ValueError:
+        raise ValueError('Ei is not a float or numpy array')
+    except TypeError:
+        if len(Ei) != len_hkl:
+            raise ValueError('Ei has different length than hkl')
+    try:
+        # check if float
+        DeltaE = float(DeltaE)
+        DeltaE = np.full(len_hkl, DeltaE)
+    except ValueError:
+        raise ValueError('DeltaE is not a float or numpy array')
+    except TypeError:
+        if len(DeltaE) != len_hkl:
+            raise ValueError('DeltaE has different length than hkl')
+
+    try:
+        # check if int
+        sign = int(sign)
+        sign = np.full(len_hkl,sign)
+    except ValueError:
+        raise ValueError('sign is not an int or numpy array')
+    except TypeError:
+        if len(sign) != len_hkl:
+            raise ValueError('sign has different length than hkl')
+
+    error_code = np.full((len_hkl), ErrorCodes.CORRECT)
+
+    try:
+        UB = lattice.getUB() * 2. * np.pi
+    except:
+        raise ValueError("Can't get the UB matrix from the lattice object")
+
+    # inputs for geometry and goniometer constraints
+    if detector_constraints:
+        if horizontal_extent[0]<-180 or horizontal_extent[1]<horizontal_extent[0] or horizontal_extent[1]>180:
+            raise ValueError("Horizontal constraints must obey -180 <= horizontal_extent[0] <= "
+                             "horizontal_extent[1] <=180")
+        if vertical_extent[0]<-180 or vertical_extent[1]<vertical_extent[0] or vertical_extent[1]>180:
+            raise ValueError("Vertical constraints must obey -180 <= vertical_extent[0] <= "
+                             "vertical_extent[1] <=180")
+        if horizontal_extent_low[0]<-180 or horizontal_extent_low[1]<horizontal_extent_low[0] or horizontal_extent_low[1]>180:
+            raise ValueError("Horizontal constraints must obey -180 <= horizontal_extent_low[0] <= "
+                             "horizontal_extent_low[1] <=180")
+        if vertical_extent_low[0]<-180 or vertical_extent_low[1]<vertical_extent_low[0] or vertical_extent_low[1]>180:
+            raise ValueError("Vertical constraints must obey -180 <= vertical_extent_low[0] <= "
+                             "vertical_extent_low[1] <=180")
+
+    if goniometer_constraints:
+        if goniometer_range[1]<goniometer_range[0] or \
+           goniometer_range[0]<-180. or goniometer_range[1]>180.:
+            raise ValueError("goniometer_range must be an increasing array, "
+                             "with both limits between -180 and 180 degrees")
```

Comment:
There are about 50 lines of codes to check inputs.  Do you consider to put them in a separate method?

Reply:
Will do.